File: backend/app/schemas/aquifer.py
"""Aquifer Pydantic schemas."""
import uuid
from datetime import datetime
from typing import Optional, Any
from pydantic import BaseModel
from app.models.aquifer import AquiferType


class AquiferBase(BaseModel):
    name: str
    type: AquiferType
    block_id: Optional[uuid.UUID] = None
    min_depth: Optional[float] = None
    max_depth: Optional[float] = None
    porosity: Optional[float] = None
    hydraulic_conductivity: Optional[float] = None
    transmissivity: Optional[float] = None
    storage_coefficient: Optional[float] = None
    specific_yield: Optional[float] = None
    quality_ec: Optional[float] = None
    dtw_decadal_avg: Optional[float] = None
    fractures_encountered: Optional[str] = None
    yield_range: Optional[str] = None


class AquiferCreate(AquiferBase):
    geometry: Optional[Any] = None


class AquiferUpdate(BaseModel):
    name: Optional[str] = None
    type: Optional[AquiferType] = None
    block_id: Optional[uuid.UUID] = None
    geometry: Optional[Any] = None
    min_depth: Optional[float] = None
    max_depth: Optional[float] = None
    porosity: Optional[float] = None
    hydraulic_conductivity: Optional[float] = None
    transmissivity: Optional[float] = None
    storage_coefficient: Optional[float] = None
    specific_yield: Optional[float] = None
    quality_ec: Optional[float] = None
    dtw_decadal_avg: Optional[float] = None
    fractures_encountered: Optional[str] = None
    yield_range: Optional[str] = None


class AquiferResponse(AquiferBase):
    id: uuid.UUID
    thickness: Optional[float] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/aquifer.ts
import type { GeoJSONGeometry } from './common';

export type AquiferType = 'unconfined' | 'confined' | 'semi_confined' | 'perched';

export interface Aquifer {
    id: string;
    name: string;
    type: AquiferType;
    block_id: string | null;
    min_depth: number | null;
    max_depth: number | null;
    thickness: number | null;        // computed on backend
    porosity: number | null;
    hydraulic_conductivity: number | null;
    transmissivity: number | null;
    storage_coefficient: number | null;
    specific_yield: number | null;
    quality_ec: number | null;
    dtw_decadal_avg: number | null;
    fractures_encountered: string | null;
    yield_range: string | null;
    geometry: GeoJSONGeometry | null;
    created_at: string;
    updated_at: string;
}

export interface AquiferCreate {
    name: string;
    type: AquiferType;
    block_id?: string;
    min_depth?: number;
    max_depth?: number;
    porosity?: number;
    hydraulic_conductivity?: number;
    transmissivity?: number;
    storage_coefficient?: number;
    specific_yield?: number;
    quality_ec?: number;
    dtw_decadal_avg?: number;
    fractures_encountered?: string;
    yield_range?: string;
    geometry?: GeoJSONGeometry;
}

export interface AquiferUpdate extends Partial<AquiferCreate> { }
